From fcf9ff2ac56c47d67fbc722575704e47265036d1 Mon Sep 17 00:00:00 2001
From: Daniel Axtens <dja@axtens.net>
Date: Mon, 6 Sep 2021 15:46:13 +1000
Subject: [PATCH 026/261] powerpc: Drop Open Hack'Ware - remove
 GRUB_IEEE1275_FLAG_CANNOT_SET_COLORS

Open Hack'Ware was the only user.

Signed-off-by: Daniel Axtens <dja@axtens.net>
Reviewed-by: Daniel Kiper <daniel.kiper@oracle.com>
---
 grub-core/term/ieee1275/console.c | 15 ++++++---------
 grub-core/video/ieee1275.c        |  7 +++----
 include/grub/ieee1275/ieee1275.h  |  3 ---
 3 files changed, 9 insertions(+), 16 deletions(-)

diff --git a/grub-core/term/ieee1275/console.c b/grub-core/term/ieee1275/console.c
index 7e797a7d4..87cbc7a1e 100644
--- a/grub-core/term/ieee1275/console.c
+++ b/grub-core/term/ieee1275/console.c
@@ -171,6 +171,7 @@ static grub_err_t
 grub_console_init_output (struct grub_term_output *term)
 {
   grub_ssize_t actual;
+  unsigned int col;
 
   /* The latest PowerMacs don't actually initialize the screen for us, so we
    * use this trick to re-open the output device (but we avoid doing this on
@@ -184,16 +185,12 @@ grub_console_init_output (struct grub_term_output *term)
     return grub_error (GRUB_ERR_UNKNOWN_DEVICE, "cannot find stdout");
 
   /* Initialize colors.  */
-  if (! grub_ieee1275_test_flag (GRUB_IEEE1275_FLAG_CANNOT_SET_COLORS))
-    {
-      unsigned col;
-      for (col = 0; col < ARRAY_SIZE (colors); col++)
-	grub_ieee1275_set_color (stdout_ihandle, col, colors[col].red,
-				 colors[col].green, colors[col].blue);
+  for (col = 0; col < ARRAY_SIZE (colors); col++)
+    grub_ieee1275_set_color (stdout_ihandle, col, colors[col].red,
+			     colors[col].green, colors[col].blue);
 
-      /* Set the right fg and bg colors.  */
-      grub_terminfo_setcolorstate (term, GRUB_TERM_COLOR_NORMAL);
-    }
+  /* Set the right fg and bg colors. */
+  grub_terminfo_setcolorstate (term, GRUB_TERM_COLOR_NORMAL);
 
   grub_console_dimensions ();
 
diff --git a/grub-core/video/ieee1275.c b/grub-core/video/ieee1275.c
index 17a3dbbb5..f437fb0df 100644
--- a/grub-core/video/ieee1275.c
+++ b/grub-core/video/ieee1275.c
@@ -93,10 +93,9 @@ grub_video_ieee1275_init (void)
 
   grub_memset (&framebuffer, 0, sizeof(framebuffer));
 
-  if (! grub_ieee1275_test_flag (GRUB_IEEE1275_FLAG_CANNOT_SET_COLORS)
-      && !grub_ieee1275_get_integer_property (grub_ieee1275_chosen,
-					      "stdout", &stdout_ihandle,
-					      sizeof (stdout_ihandle), &actual)
+  if (!grub_ieee1275_get_integer_property (grub_ieee1275_chosen,
+					   "stdout", &stdout_ihandle,
+					   sizeof (stdout_ihandle), &actual)
       && actual == sizeof (stdout_ihandle))
     have_setcolors = 1;
 
diff --git a/include/grub/ieee1275/ieee1275.h b/include/grub/ieee1275/ieee1275.h
index 9d410d821..8a1206195 100644
--- a/include/grub/ieee1275/ieee1275.h
+++ b/include/grub/ieee1275/ieee1275.h
@@ -105,9 +105,6 @@ enum grub_ieee1275_flag
   /* OLPC / XO firmware hangs when accessing USB devices.  */
   GRUB_IEEE1275_FLAG_OFDISK_SDCARD_ONLY,
 
-  /* Open Hack'Ware stops when trying to set colors */
-  GRUB_IEEE1275_FLAG_CANNOT_SET_COLORS,
-
   /* Open Hack'Ware stops when grub_ieee1275_interpret is used.  */
   GRUB_IEEE1275_FLAG_CANNOT_INTERPRET,
 
-- 
2.36.1

